### PR TITLE
Add 6.8 to the remote clusters compatibility table

### PR DESCRIPTION
**Pull request by javanna**
The table does not include 6.8 as it was written before we knew we were releasing it. This PR adds it. Hopefully the explanation is correct and not too confusing, as we effectively have two last minor versions for the 6.x series.

### CODE REVIEW

**Comment by javanna** [comment]
I have updated the table according to internal discussions. I am not too happy with as it's complicated to read due to 6.8 being released after 7.0, which requires separate rows for 6.7, 6.8, 7.0 and 7.1 (or later). I wish that we could simplify this but that may create even more confusion as it would make these compatibility rules diverge from the rolling upgrades rules. I could definitely use some help here :)

File: docs/reference/modules/remote-clusters.asciidoc [hunk 1]
```diff
@@ -32,22 +32,26 @@ The _gateway nodes_ selection depends on the following criteria:
...
 Any node can communicate with any other node on the same major version (e.g.
-6.0 can talk to any 6.x node). Only nodes on the last minor version of a
-certain major version can communicate with nodes on the following major
-version (e.g. 6.7 can communicate with 7.0, as well as any 7.x node, while
- 6.6 or earlier cannot talk to any 7.x node). Note that version compatibility
+6.0 can talk to any 6.x node). Only nodes on the last minor version (6.8 is a
+bit special in this context as it was released after 7.0) of a certain major
+version can communicate with nodes on the following major version: 6.7 can
+communicate with 7.0, as well as any 7.x node, while 6.6 or earlier cannot
+talk to any 7.x node; note that 7.1, which was released after 7.0, cannot
```
Comment (rjernst) [inline docs/reference/modules/remote-clusters.asciidoc]: 7.1 -> 6.8?

File: docs/reference/modules/remote-clusters.asciidoc [hunk 2]
```diff
@@ -32,22 +32,26 @@ The _gateway nodes_ selection depends on the following criteria:
...
-| 5.0->5.5      |    Yes   | Yes |    No    | No  | No
-| 5.6           |    Yes   | Yes |    Yes   | Yes | No
-| 6.0->6.6      |    No    | Yes |    Yes   | Yes | No
-| 6.7           |    No    | Yes |    Yes   | Yes | Yes
-| 7.x           |    No    | No  |    No    | Yes | Yes
+| Compatibility | 5.0->5.5 | 5.6 | 6.0->6.6 | 6.7 | 6.8 | 7.0 | 7.1->7.x
+| 5.0->5.5      |    Yes   | Yes |    No    | No  | No  | No  |    No
+| 5.6           |    Yes   | Yes |    Yes   | Yes | Yes | No  |    No
+| 6.0->6.6      |    No    | Yes |    Yes   | Yes | Yes | No  |    No
+| 6.7           |    No    | Yes |    Yes   | Yes | Yes | Yes |    Yes
```
Comment (rjernst) [inline docs/reference/modules/remote-clusters.asciidoc]: I think the 7.1->7.x column should be No here? We guarantee the last minor fo the previous major. Now that 6.8 is out, the compatible version checks in 7.1+ would see 6.8 as the minimum compatible version, not 6.7.

**Review by rjernst — COMMENTED** [review]
Thanks for tweaking this @javanna. I've been thinking on this more, and technically we do still support wire compatibility for 6.8 to 7.0. I know before I suggested keeping this simple, but it didn't occur to me the docs you were changing were only about remote clusters, in which case upgrading doesn't matter (which is where 6.8 to 7.0 breaks). I'm fine keeping this as is, but if you or others want, I would be fine with going back to what you had before (but what you have here is technically correct given our compatibility policies, minus the 2 comments I left).

File: docs/reference/modules/remote-clusters.asciidoc (hunk 2 above)
Comment (javanna) [reply docs/reference/modules/remote-clusters.asciidoc]: good to know, I was not sure.

File: docs/reference/modules/remote-clusters.asciidoc (hunk 1 above)
Comment (javanna) [reply docs/reference/modules/remote-clusters.asciidoc]: yep thanks for catching this one.

**Comment by javanna** [comment]
Thanks @rjernst for looking. I agree with you. As much as I would like to keep this aligned with rolling upgrades rules, communications between remote clusters is slightly different. I went back to something similar to what I had before. I find it simpler to read as well. Let me know what you think.

File: docs/reference/modules/remote-clusters.asciidoc [hunk 3]
```diff
@@ -30,24 +30,24 @@ node to the selected remote _gateway nodes_ only.
...
-version (e.g. 6.7 can communicate with 7.0, as well as any 7.x node, while
- 6.6 or earlier cannot talk to any 7.x node). Note that version compatibility
- is symmetric, meaning that if 6.7 can communicate with 7.0, 7.0 can also
- communicate with 6.7. The matrix below summarizes compatibility as described
- above.
+registered to. This is subject to rules that are similar to those for
+<<rolling-upgrades>>. Any node can communicate with any other node on the same
+major version (e.g. 7.0 can talk to any 7.x node). Only nodes on the last minor
+version of a certain major version can communicate with nodes on the following
+major version. Note that in the 6.x series, both 6.7 and 6.8 can communicate
```
Comment (rjernst) [inline docs/reference/modules/remote-clusters.asciidoc]: I think this is still incorrect. I don't think a 7.1 node will communicate with a 6.7 node, and we certainly don't test it. To upgrade across majors to 7.1, a user must come from 6.8. Version.minimumCompatibilityVersion() looks at the version constants and finds the previous major's last minor.

Comment (javanna) [reply docs/reference/modules/remote-clusters.asciidoc]: I tested it, and you are right 6.7 can't talk to 7.1.

**Comment by javanna** [comment]
I pushed an update @rjernst could you have another look? Hopefully the table is correct now.

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by javanna** [comment]
thanks for the help @rjernst !

**Pull request merged**